lib/screens/settings/settings_screen.dart

```diff
@@ -26,58 +26,36 @@ class SettingsScreen extends StatelessWidget {
       body: BlocProvider(
         create: (context) =>
             SettingsBloc(BlocProvider.of<AuthenticationBloc>(context))..add(const SetUpInitialValues()),
-        child: MultiBlocListener(
-          listeners: [
-            BlocListener<SettingsBloc, SettingsState>(
-              listenWhen: (_, current) => current.navigateToGuardians != null,
-              listener: (context, _) => NavigationService.of(context).navigateTo(Routes.guardianTabs),
-            ),
-            BlocListener<SettingsBloc, SettingsState>(
-              listenWhen: (_, current) => current.navigateToRecoverAccount != null,
-              listener: (context, _) => NavigationService.of(context).navigateTo(Routes.recoverAccountSearch),
-            ),
-            BlocListener<SettingsBloc, SettingsState>(
-              listenWhen: (_, current) => current.navigateToVerification != null,
-              listener: (context, _) {
-                BlocProvider.of<SettingsBloc>(context).add(const ResetNavigateToVerification());
-                NavigationService.of(context).navigateTo(Routes.verification).then((isValid) {
-                  if (isValid ?? false) {
-                    BlocProvider.of<SettingsBloc>(context).add(const OnValidVerification());
-                  }
-                });
-              },
-            ),
-            BlocListener<SettingsBloc, SettingsState>(
-              listenWhen: (previous, current) =>
-                  previous.isSecureBiometric == false && current.isSecureBiometric == true,
```

Comment:
previous.isSecureBiometric == false && current.isSecureBiometric == true,

All these checks should not be here, they should all happen in the bloc, then send page commands for the UI to react.